pkg/hwinfo/hwinfo_nonwindows.go

```diff
@@ -61,16 +63,27 @@ func retrieveInfoUsingDmiDecode() (map[string]interface{}, error) {
 		return nil, nil
 	}
 
-	// expecting /etc/sudoers.d/cagent-dmidecode is present
-	cmd := exec.Command("/bin/sh", "-c", "sudo dmidecode")
+	// expecting 'sudo' package is installed and /etc/sudoers.d/cagent-dmidecode is present
+	const dmidecodeCmd = "sudo dmidecode"
+	cmd := exec.Command("/bin/sh", "-c", dmidecodeCmd)
+
+	stdoutBuffer := bytes.Buffer{}
+	cmd.Stdout = bufio.NewWriter(&stdoutBuffer)
+
+	stderrBuffer := bytes.Buffer{}
+	cmd.Stderr = bufio.NewWriter(&stderrBuffer)
 
-	buf := bytes.Buffer{}
-	cmd.Stdout = bufio.NewWriter(&buf)
 	if err := cmd.Run(); err != nil {
+		stderrBytes, _ := ioutil.ReadAll(bufio.NewReader(&stderrBuffer))
+		stderr := string(stderrBytes)
+		if strings.Contains(stderr, "/dev/mem: Operation not permitted") {
+			log.Infof("[HWINFO] there was an error while executing '%s': %s\nProbably 'CONFIG_STRICT_DEVMEM' kernel configuration option is enabled. Please refer to kernel configuration manual.", dmidecodeCmd, stderr)
```

Comment:
Is this `Info` log level and not rather `Warning` or `Error`?

Reply:
Thorsten and I think `Info` is more appropriate because it's not a failure of cagent neither an error of system configuration. It's just info for the user that some data couldn't be acquired because the kernel configuration does not allow it.

Also, we use `info`  log level for similar cases in `hwinfo` package and others.